go-controller/pkg/node/openflow_manager_test.go

```diff
@@ -0,0 +1,66 @@
+package node
+
+import "testing"
+
+func TestOpenFlowManagerDefaultNetOVSBridgeFinder(t *testing.T) {
+	testCases := []struct {
+		name                  string
+		desc                  string
+		inputPortInfo         string
+		expectedBridgeName    string
+		expectedPatchPortName string
+	}{
+		{
+			name:                  "empty input ports",
+			inputPortInfo:         "",
+			expectedBridgeName:    "",
+			expectedPatchPortName: "",
+		},
+		{
+			name:                  "input ports without patch ports",
+			inputPortInfo:         "port1",
+			expectedBridgeName:    "",
+			expectedPatchPortName: "",
+		},
+		{
+			name: "input ports with a patch port",
+			inputPortInfo: `
+port1
+port2
+patch-br-ex_multi-homing-worker-0.maiqueb.org-to-br-int`,
+			expectedBridgeName:    "br-ex",
+			expectedPatchPortName: "patch-br-ex_multi-homing-worker-0.maiqueb.org-to-br-int",
+		},
+		{
+			name: "input ports with a patch port for a localnet network",
+			inputPortInfo: `
+port1
+port2
+patch-vlan2003_ovn_localnet_port-to-br-int`,
+			expectedBridgeName:    "",
+			expectedPatchPortName: "",
+		},
+		{
+			name: "input ports with a patch port for the default network and a localnet",
+			inputPortInfo: `
+port1
```

Comment:
usually is this function also parsing by new line ? is localnetInfo expecting a list of ports or a single port... ?

Reply:
list of ports. It is fed pretty much the output of `ovs-vsctl list interfaces` (just the name).

The command that extracts the port names outputs them separated by `\n`.